### PR TITLE
[bsp][RA6M4]Add segger rtt function

### PR DESCRIPTION
**Pull request by kurisaW**
## 拉取/合并请求描述：(PR description)

[ 添加segger rtt 为console的Kconfig]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [ ] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](../documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/contribution_guide/coding_style_en.txt)

### CODE REVIEW

**Review by kurisaW — COMMENTED** [review]
[modify] 格式对齐修改

**Review by mysterywolf — APPROVED** [review]

File: bsp/renesas/libraries/HAL_Drivers/drv_common.c [hunk 1]
```diff
@@ -130,6 +130,11 @@ RT_WEAK void rt_hw_board_init()
 #endif
 
     /* Set the shell console output device */
+#if defined(SEGGER_RTT_ENABLE)
+extern int rt_hw_jlink_rtt_init(void);
+    rt_hw_jlink_rtt_init();
+    rt_console_set_device("jlinkRtt");
+#else
 #if defined(RT_USING_CONSOLE) && defined(RT_USING_DEVICE)
```
Comment (mysterywolf) [inline bsp/renesas/libraries/HAL_Drivers/drv_common.c:138]:
```suggestion
#elif defined(RT_USING_CONSOLE) && defined(RT_USING_DEVICE)
```

**Review by mysterywolf — CHANGES_REQUESTED** [review]

**Review by kurisaW — COMMENTED** [review]
[modify]

**Review by kurisaW — COMMENTED** [review]
[modify]

**Review by kurisaW — COMMENTED** [review]
[modify]

**Review by mysterywolf — APPROVED** [review]

File: bsp/renesas/libraries/HAL_Drivers/drv_common.c [hunk 2]
```diff
@@ -130,7 +130,11 @@ RT_WEAK void rt_hw_board_init()
 #endif
 
     /* Set the shell console output device */
-#if defined(RT_USING_CONSOLE) && defined(RT_USING_DEVICE)
+#if defined(SEGGER_RTT_ENABLE)
```
Comment (supperthomas) [inline bsp/renesas/libraries/HAL_Drivers/drv_common.c:133]: 这里可以这样用了？试过了么？可以打印启动的log么？如果有问题这里可以先不加。

**Review by supperthomas — CHANGES_REQUESTED** [review]

File: bsp/renesas/libraries/HAL_Drivers/drv_common.c (hunk 2 above)
Comment (kurisaW) [reply bsp/renesas/libraries/HAL_Drivers/drv_common.c]: 测试过不能使用，已经将此处删除，能正常运行

**Review by kurisaW — COMMENTED** [review]
[modify]

**Review by supperthomas — APPROVED** [review]

**Comment by mysterywolf** [comment]
https://github.com/RT-Thread/rt-thread/blob/fc55785e70fbf8d117b7f713f67131f640e9fe3d/bsp/nrf5x/nrf52840/board/board.c
这个是否也需要删除？ @supperthomas

**Comment by supperthomas** [comment]
> https://github.com/RT-Thread/rt-thread/blob/fc55785e70fbf8d117b7f713f67131f640e9fe3d/bsp/nrf5x/nrf52840/board/board.c
> 这个是否也需要删除？ @supperthomas 

不用，两个串口框架不一样，那个可以用，这个用不了

**Comment by supperthomas** [comment]
https://github.com/RT-Thread/rt-thread/issues/6062

**Pull request merged**